### PR TITLE
docs(forwarding-refs.md): add warning to third-party library maintainers

### PR DESCRIPTION
**Pull request by jedwards1211**
@gaearon what do you think of this message?

### CODE REVIEW

File: content/docs/forwarding-refs.md [hunk 1]
```diff
@@ -37,6 +37,12 @@ Here is a step-by-step explanation of what happens in the above example:
 >
 >Ref forwarding is not limited to DOM components. You can forward refs to class component instances, too.
 
+## Warning for third-party library maintainers
+
+**If you start using `forwardRef` in a third-party library, you should release it as a breaking change.**.  This is because the value returned by `forwardRef` is of a different type than React components have historically been (a `class` extending `React.Component` or a `function`) and many HoC libraries (e.g. `react-redux`' `connect()`) aren't currently compatible with such components.
```
Comment (gaearon) [inline content/docs/forwarding-refs.md]: I would just leave it at "has an observably different behavior (including the component type and what the ref gets assigned to), and can break apps and other libraries that depend on the old behavior." I don't think there's a need for jargon like "HoC" or mentioning specific libraries.

File: content/docs/forwarding-refs.md [hunk 2]
```diff
@@ -37,6 +37,12 @@ Here is a step-by-step explanation of what happens in the above example:
...
+
+Even if you check whether `React.forwardRef` is defined and only use it if it is present, you should release a breaking change, because it may still cause problems for users of your library.
```
Comment (gaearon) [inline content/docs/forwarding-refs.md]: We can just say "Conditionally applying `forwardRef` when it exists is also not recommended for the same reasons: it changes how your library behaves and can break your users' apps as they upgrade React versions."

File: content/docs/forwarding-refs.md (hunk 1 above)
Comment (jedwards1211) [reply content/docs/forwarding-refs.md]: okay, sounds good

**Comment by jedwards1211** [comment]
@gaearon okay, I made your suggested changes

File: content/docs/forwarding-refs.md [hunk 3]
```diff
@@ -37,6 +37,12 @@ Here is a step-by-step explanation of what happens in the above example:
 >
 >Ref forwarding is not limited to DOM components. You can forward refs to class component instances, too.
 
+## Warning for third-party library maintainers
```
Comment (gaearon) [inline content/docs/forwarding-refs.md]: Maybe "for component library maintainers" to be clearer

**Comment by gaearon** [comment]
thanks

**Pull request merged**